usvm-ts/src/test/kotlin/org/usvm/util/TSMethodTestRunner.kt

```diff
@@ -0,0 +1,178 @@
+
+    protected inline fun <reified T1, reified T2, reified R> discoverProperties(
+        methodIdentifier: MethodDescriptor,
+        vararg analysisResultMatchers: (T1, T2, R?) -> Boolean,
+        invariants: Array<out Function<Boolean>> = emptyArray(),
+    ) {
+        internalCheck(
+            target = methodIdentifier,
+            analysisResultsNumberMatcher = ignoreNumberOfAnalysisResults,
+            analysisResultsMatchers = analysisResultMatchers,
+            invariants = invariants,
+            extractValuesToCheck = { r -> r.parameters + r.resultValue },
+            expectedTypesForExtractedValues = arrayOf(
+                typeTransformer(T1::class),
+                typeTransformer(T2::class),
+                typeTransformer(R::class)
+            ),
+            checkMode = CheckMode.MATCH_PROPERTIES,
+            coverageChecker = { _ -> true }
+        )
+    }
+
+    protected inline fun <reified T1, reified T2, reified T3, reified R> discoverProperties(
+        methodIdentifier: MethodDescriptor,
+        vararg analysisResultMatchers: (T1, T2, T3, R?) -> Boolean,
+        invariants: Array<out Function<Boolean>> = emptyArray(),
+    ) {
+        internalCheck(
+            target = methodIdentifier,
+            analysisResultsNumberMatcher = ignoreNumberOfAnalysisResults,
+            analysisResultsMatchers = analysisResultMatchers,
+            invariants = invariants,
+            extractValuesToCheck = { r -> r.parameters + r.resultValue },
+            expectedTypesForExtractedValues = arrayOf(
+                typeTransformer(T1::class),
+                typeTransformer(T2::class),
+                typeTransformer(T3::class),
+                typeTransformer(R::class)
+            ),
+            checkMode = CheckMode.MATCH_PROPERTIES,
+            coverageChecker = { _ -> true }
+        )
+    }
+
+    protected inline fun <reified T1, reified T2, reified T3, reified T4, reified R> discoverProperties(
+        methodIdentifier: MethodDescriptor,
+        vararg analysisResultMatchers: (T1, T2, T3, T4, R?) -> Boolean,
+        invariants: Array<out Function<Boolean>> = emptyArray(),
+    ) {
+        internalCheck(
+            target = methodIdentifier,
+            analysisResultsNumberMatcher = ignoreNumberOfAnalysisResults,
+            analysisResultsMatchers = analysisResultMatchers,
+            invariants = invariants,
+            extractValuesToCheck = { r -> r.parameters + r.resultValue },
+            expectedTypesForExtractedValues = arrayOf(
+                typeTransformer(T1::class),
+                typeTransformer(T2::class),
+                typeTransformer(T3::class),
+                typeTransformer(T4::class),
+                typeTransformer(R::class)
+            ),
+            checkMode = CheckMode.MATCH_PROPERTIES,
+            coverageChecker = { _ -> true }
+        )
+    }
+
+    override val typeTransformer: (Any?) -> EtsType
+        get() = TODO("Not yet implemented")
```

Comment:
Is it ok?

Reply:
Yes, all tests for TS are disabled now and require additional work to enable. It will be done in further requests